sympy/discrete/recurrences.py

```diff
@@ -6,7 +6,7 @@
 from sympy.core import S, sympify
 from sympy.core.compatibility import as_int, iterable
 
-def linrec(coeffs, init, n):
+def linrec(coeffs, init, n, final_expr=False):
```

Comment:
How about having a separate `linrec_coeffs(coeffs, n)` that could also be used by `linrec`?

Reply:
Yes, I would also agree on @jksuom 's idea to have a separate function than adding up a keyword.
It is not a good design because setting the keyword `True` would make the variable `init` obscure.

Reply:
can you look at the error?

Reply:
got it